src/components/AppointmentCard/AppointmentCardContainer.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { AppointmentCard } from 'components/AppointmentCard/AppointmentCard';
+import { AppointmentCardContainerProps } from 'components/AppointmentCard/AppointmentCard.types';
+import { ROLES } from 'routes/constants';
+import { ReactComponent as Board } from '../../assets/svg/board-icon.svg';
+import { ReactComponent as Heart } from '../../assets/svg/heart-icon.svg';
+
+export const AppointmentCardContainer:React.VFC<AppointmentCardContainerProps> = ({
+  photo,
+  appointmentID,
+  firstName,
+  lastName,
+  status,
+  role,
+  specialization,
+  resolution,
+  reason,
+  time,
+}) => {
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+  const menuRef = useRef() as React.MutableRefObject<HTMLDivElement> | undefined;
+  const cardDescription = reason || resolution?.resolution;
+  const cardIcon = role === ROLES.DOCTOR ? <Board /> : <Heart />;
```

Comment:
Could be moved to props and passed from explicit role container as component for render